File: ios/CreatePrivateDealerOS/CreatePrivateDealerOS/Networking/RemoteDataService.swift
import Foundation

protocol RemoteDataService {
    func fetchUserProfile(userID: UUID, token: String) async throws -> UserProfile
    func fetchDealerSettings(dealerID: UUID, token: String) async throws -> DealerSettings
    func updateDealerSettings(dealerID: UUID, settings: DealerSettings, token: String) async throws

    func fetchDashboardMetrics(user: UserProfile, missedLeadMinutes: Int, token: String) async throws -> DashboardMetrics
    func fetchConversations(dealerID: UUID, token: String) async throws -> [Conversation]
    func fetchMessages(conversationID: UUID, dealerID: UUID, token: String) async throws -> [Message]
    func sendMessage(payload: SendMessagePayload, token: String) async throws
    func handoff(payload: HandoffPayload, token: String) async throws

    func fetchLeads(dealerID: UUID, filter: LeadFilter, token: String) async throws -> [Lead]
    func updateLeadStatus(leadID: UUID, dealerID: UUID, status: LeadStatus, token: String) async throws
    func assignLead(leadID: UUID, dealerID: UUID, userID: UUID, token: String) async throws
    func fetchFollowups(leadID: UUID, dealerID: UUID, token: String) async throws -> [FollowupItem]

    func fetchBookings(dealerID: UUID, token: String) async throws -> [Booking]
    func createBooking(_ request: NewBookingRequest, token: String) async throws
    func updateBookingStatus(bookingID: UUID, dealerID: UUID, status: BookingStatus, token: String) async throws

    func fetchTeamMembers(dealerID: UUID, token: String) async throws -> [TeamMember]
    func inviteUser(_ request: InviteUserRequest, token: String) async throws
    func disableUser(userID: UUID, dealerID: UUID, token: String) async throws

    func fetchAnalytics(dealerID: UUID, token: String) async throws -> AnalyticsSnapshot
    func registerDeviceToken(_ deviceToken: String, token: String) async
}

struct LeadFilter: Hashable {
    var status: LeadStatus?
    var source: LeadSource?
    var assignedUserID: UUID?
}

struct FollowupItem: Codable, Identifiable, Hashable {
    let id: UUID
    let dealerID: UUID
    let leadID: UUID
    let type: String
    let sentVia: String
    let sentAt: Date?
    let responded: Bool
    let responseAt: Date?
    let createdAt: Date
}

File: flutter/create_private_dealer_os/lib/data/services/remote_data_service.dart
import '../../domain/models/analytics_snapshot.dart';
import '../../domain/models/booking.dart';
import '../../domain/models/conversation.dart';
import '../../domain/models/dashboard_metrics.dart';
import '../../domain/models/followup_item.dart';
import '../../domain/models/lead.dart';
import '../../domain/models/settings.dart';
import '../../domain/models/user_profile.dart';

abstract class RemoteDataService {
  Future<UserProfile> fetchUserProfile({
    required String userId,
    required String token,
  });

  Future<DealerSettings> fetchDealerSettings({
    required String dealerId,
    required String token,
  });

  Future<void> updateDealerSettings({
    required String dealerId,
    required DealerSettings settings,
    required String token,
  });

  Future<DashboardMetrics> fetchDashboardMetrics({
    required UserProfile user,
    required int missedLeadMinutes,
    required String token,
  });

  Future<List<Conversation>> fetchConversations({
    required String dealerId,
    required String token,
  });

  Future<List<Message>> fetchMessages({
    required String conversationId,
    required String dealerId,
    required String token,
  });

  Future<void> sendMessage({
    required String conversationId,
    required String leadId,
    required String content,
    required String token,
  });

  Future<void> handoff({
    required String conversationId,
    required String leadId,
    required String reason,
    required String token,
  });

  Future<List<Lead>> fetchLeads({
    required String dealerId,
    required LeadFilter filter,
    required String token,
  });

  Future<void> updateLeadStatus({
    required String leadId,
    required String dealerId,
    required LeadStatus status,
    required String token,
  });

  Future<void> assignLead({
    required String leadId,
    required String dealerId,
    required String userId,
    required String token,
  });

  Future<List<FollowupItem>> fetchFollowups({
    required String leadId,
    required String dealerId,
    required String token,
  });

  Future<List<Booking>> fetchBookings({
    required String dealerId,
    required String token,
  });

  Future<void> createBooking({
    required NewBookingRequest request,
    required String token,
  });

  Future<void> updateBookingStatus({
    required String bookingId,
    required String dealerId,
    required BookingStatus status,
    required String token,
  });

  Future<List<TeamMember>> fetchTeamMembers({
    required String dealerId,
    required String token,
  });

  Future<void> inviteUser({required InviteUserRequest request, required String token});

  Future<void> disableUser({
    required String userId,
    required String dealerId,
    required String token,
  });

  Future<AnalyticsSnapshot> fetchAnalytics({
    required String dealerId,
    required String token,
  });

  Future<void> registerDeviceToken({
    required String deviceToken,
    required String token,
  });
}
